Saule/Queries/Pagination/IPagedResult.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections;
+
+namespace Saule.Queries.Pagination
+{
+    /// <summary>
+    /// Paged result that can be used to tell Saule how many pages we have
+    /// </summary>
+    public interface IPagedResult
+    {
+        /// <summary>
+        /// Gets total count of items across all pages
+        /// </summary>
+        int TotalResultsCount { get; }
+
+        /// <summary>
+        /// Gets data items
+        /// </summary>
+        IEnumerable Items { get; }
```

Comment:
Not a big fan of the untyped enumerable. It also blocks people from doing client-powered filtering on database level using `IQueryable`. I think we need to put some thought into all the use-cases here and make sure we handle them, I think.

Reply:
I'll also add implementation for IQueryable case too

Reply:
Implemented handling for Count for IQueryable\IEnumerable (as we already did it for Skip\Take for both interfaces)